lib/src/micheline_decoder/impl/micheline_decoder.dart

```diff
@@ -0,0 +1,79 @@
+import 'address_decoder.dart';
+import 'big_map_decoder.dart';
+import 'bytes_decoder.dart';
+import 'int_decoder.dart';
+import 'key_decoder.dart';
+import 'list_decoder.dart';
+import 'map_decoder.dart';
+import 'option_decoder.dart';
+import 'pair_decoder.dart';
+import 'signature_decoder.dart';
+import 'string_decoder.dart';
+import 'timestamp_decoder.dart';
+
+class MichelineDecoder {
+  final Map<String, dynamic> schema;
+  final data;
+
+  MichelineDecoder({required this.schema, required this.data});
+
+  dynamic decode() {
+    final prim = schema['prim'];
+    MichelineDecoder decoder;
+
+    switch (prim) {
```

Comment:
Pas obligatoire mais le switch pourrait être factorisé dans une Factory (https://scottt2.github.io/design-patterns-in-dart/factory_method/).

Reply:
Oui bien vu